python/recipe.cpp

```diff
@@ -24,6 +24,7 @@ namespace pyarb {
 // The py::recipe::cell_decription returns a pybind11::object, that is
 // unwrapped and copied into a arb::util::unique_any.
 arb::util::unique_any py_recipe_shim::get_cell_description(arb::cell_gid_type gid) const {
+    pybind11::gil_scoped_acquire guard;
```

Comment:
Why does this fix the problem with temporaries?

`try_catch_pyexception` uses a global mutex that we use on the C++ side, and `convert_cell` acquires the GIL explicitly. Acquiring the GIL at this level could lead to deadlock.

Reply:
Quoting `cells.cpp` comment right before `convert_cells`
```
// Warning: requires that the GIL has been acquired before calling,
// if there is a segmentation fault in the cast or isinstance calls,
// check that the caller has the GIL.
arb::util::unique_any convert_cell(pybind11::object o) {
```

Reply:
Retracted.